// @sondh0127/vue-epg-planby/dist/index.es.js
import { ref as S, defineComponent as q, openBlock as p, createElementBlock as M, normalizeStyle as T, unref as l, createElementVNode as b, createTextVNode as vt, createBlock as ee, createCommentVNode as Q, renderSlot as U, createVNode as gt, toRefs as yt, computed as P, watchEffect as Be, toRef as ge, Fragment as ye, renderList as pe, normalizeProps as I, mergeProps as je, toDisplayString as te, guardReactiveProps as G, withCtx as oe, useCssVars as Ae, normalizeClass as pt, onMounted as wt, onBeforeUnmount as bt } from "vue";
import { computed as Tt, ref as Ve, isRef as Xe, watch as Ge, getCurrentScope as Dt, onScopeDispose as St, customRef as xt, unref as se, provide as Ct, inject as Mt, isVue3 as Pt } from "vue-demi";
const Ot = 7200, Wt = 24, Qe = 60, kt = 60, Et = 100, Pe = 80, _t = Pe, $t = 100, Ut = 100, ze = 12e4, va = {
  DARK: "dark",
  LIGHT: "light"
}, re = {
  DEFAULT: "yyyy-MM-dd HH:mm:ss",
  DATE: "yyyy-MM-dd",
  HOURS_MIN: "HH:mm",
  BASE_HOURS_TIME: "h:mm a"
};
function E(e) {
  if (e === null || e === !0 || e === !1)
    return NaN;
  var t = Number(e);
  return isNaN(t) ? t : t < 0 ? Math.ceil(t) : Math.floor(t);
}
function g(e, t) {
  if (t.length < e)
    throw new TypeError(e + " argument" + (e > 1 ? "s" : "") + " required, but only " + t.length + " present");
}
function x(e) {
  g(1, arguments);
  var t = Object.prototype.toString.call(e);
  return e instanceof Date || typeof e == "object" && t === "[object Date]" ? new Date(e.getTime()) : typeof e == "number" || t === "[object Number]" ? new Date(e) : ((typeof e == "string" || t === "[object String]") && typeof console < "u" && (console.warn("Starting with v2.0.0-beta.1 date-fns doesn't accept strings as date arguments. Please use `parseISO` to parse strings. See: https://git.io/fjule"), console.warn(new Error().stack)), new Date(NaN));
}
function Je(e, t) {
  g(2, arguments);
  var r = x(e), n = E(t);
  return isNaN(n) ? new Date(NaN) : (n && r.setDate(r.getDate() + n), r);
}
function Nt(e, t) {
  g(2, arguments);
  var r = x(e).getTime(), n = E(t);
  return new Date(r + n);
}
function qt(e) {
  var t = new Date(Date.UTC(e.getFullYear(), e.getMonth(), e.getDate(), e.getHours(), e.getMinutes(), e.getSeconds(), e.getMilliseconds()));
  return t.setUTCFullYear(e.getFullYear()), e.getTime() - t.getTime();
}
function L(e) {
  g(1, arguments);
  var t = x(e);
  return t.setHours(0, 0, 0, 0), t;
}
var Ht = 6e4, Ft = 36e5;
function Yt(e, t) {
  g(2, arguments);
  var r = L(e), n = L(t);
  return r.getTime() === n.getTime();
}
function Rt(e) {
  return g(1, arguments), e instanceof Date || typeof e == "object" && Object.prototype.toString.call(e) === "[object Date]";
}
function It(e) {
  if (g(1, arguments), !Rt(e) && typeof e != "number")
    return !1;
  var t = x(e);
  return !isNaN(Number(t));
}
function Ke(e, t) {
  return g(2, arguments), x(e).getTime() - x(t).getTime();
}
var ke = {
  ceil: Math.ceil,
  round: Math.round,
  floor: Math.floor,
  trunc: function(e) {
    return e < 0 ? Math.ceil(e) : Math.floor(e);
  }
}, Lt = "trunc";
function Ze(e) {
  return e ? ke[e] : ke[Lt];
}
function Ee(e, t, r) {
  g(2, arguments);
  var n = Ke(e, t) / Ft;
  return Ze(r == null ? void 0 : r.roundingMethod)(n);
}
function fe(e, t, r) {
  g(2, arguments);
  var n = Ke(e, t) / Ht;
  return Ze(r == null ? void 0 : r.roundingMethod)(n);
}
var Bt = {
  lessThanXSeconds: {
    one: "less than a second",
    other: "less than {{count}} seconds"
  },
  xSeconds: {
    one: "1 second",
    other: "{{count}} seconds"
  },
  halfAMinute: "half a minute",
  lessThanXMinutes: {
    one: "less than a minute",
    other: "less than {{count}} minutes"
  },
  xMinutes: {
    one: "1 minute",
    other: "{{count}} minutes"
  },
  aboutXHours: {
    one: "about 1 hour",
    other: "about {{count}} hours"
  },
  xHours: {
    one: "1 hour",
    other: "{{count}} hours"
  },
  xDays: {
    one: "1 day",
    other: "{{count}} days"
  },
  aboutXWeeks: {
    one: "about 1 week",
    other: "about {{count}} weeks"
  },
  xWeeks: {
    one: "1 week",
    other: "{{count}} weeks"
  },
  aboutXMonths: {
    one: "about 1 month",
    other: "about {{count}} months"
  },
  xMonths: {
    one: "1 month",
    other: "{{count}} months"
  },
  aboutXYears: {
    one: "about 1 year",
    other: "about {{count}} years"
  },
  xYears: {
    one: "1 year",
    other: "{{count}} years"
  },
  overXYears: {
    one: "over 1 year",
    other: "over {{count}} years"
  },
  almostXYears: {
    one: "almost 1 year",
    other: "almost {{count}} years"
  }
}, jt = function(e, t, r) {
  var n, a = Bt[e];
  return typeof a == "string" ? n = a : t === 1 ? n = a.one : n = a.other.replace("{{count}}", t.toString()), r != null && r.addSuffix ? r.comparison && r.comparison > 0 ? "in " + n : n + " ago" : n;
};
const At = jt;
function xe(e) {
  return function() {
    var t = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : {}, r = t.width ? String(t.width) : e.defaultWidth, n = e.formats[r] || e.formats[e.defaultWidth];
    return n;
  };
}
var Vt = {
  full: "EEEE, MMMM do, y",
  long: "MMMM do, y",
  medium: "MMM d, y",
  short: "MM/dd/yyyy"
}, Xt = {
  full: "h:mm:ss a zzzz",
  long: "h:mm:ss a z",
  medium: "h:mm:ss a",
  short: "h:mm a"
}, Gt = {
  full: "{{date}} 'at' {{time}}",
  long: "{{date}} 'at' {{time}}",
  medium: "{{date}}, {{time}}",
  short: "{{date}}, {{time}}"
}, Qt = {
  date: xe({
    formats: Vt,
    defaultWidth: "full"
  }),
  time: xe({
    formats: Xt,
    defaultWidth: "full"
  }),
  dateTime: xe({
    formats: Gt,
    defaultWidth: "full"
  })
};
const zt = Qt;
var Jt = {
  lastWeek: "'last' eeee 'at' p",
  yesterday: "'yesterday at' p",
  today: "'today at' p",
  tomorrow: "'tomorrow at' p",
  nextWeek: "eeee 'at' p",
  other: "P"
}, Kt = function(e, t, r, n) {
  return Jt[e];
};
const Zt = Kt;
function ne(e) {
  return function(t, r) {
    var n = r || {}, a = n.context ? String(n.context) : "standalone", i;
    if (a === "formatting" && e.formattingValues) {
      var o = e.defaultFormattingWidth || e.defaultWidth, s = n.width ? String(n.width) : o;
      i = e.formattingValues[s] || e.formattingValues[o];
    } else {
      var u = e.defaultWidth, c = n.width ? String(n.width) : e.defaultWidth;
      i = e.values[c] || e.values[u];
    }
    var d = e.argumentCallback ? e.argumentCallback(t) : t;
    return i[d];
  };
}
var er = {
  narrow: ["B", "A"],
  abbreviated: ["BC", "AD"],
  wide: ["Before Christ", "Anno Domini"]
}, tr = {
  narrow: ["1", "2", "3", "4"],
  abbreviated: ["Q1", "Q2", "Q3", "Q4"],
  wide: ["1st quarter", "2nd quarter", "3rd quarter", "4th quarter"]
}, rr = {
  narrow: ["J", "F", "M", "A", "M", "J", "J", "A", "S", "O", "N", "D"],
  abbreviated: ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"],
  wide: ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"]
}, nr = {
  narrow: ["S", "M", "T", "W", "T", "F", "S"],
  short: ["Su", "Mo", "Tu", "We", "Th", "Fr", "Sa"],
  abbreviated: ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"],
  wide: ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"]
}, ar = {
  narrow: {
    am: "a",
    pm: "p",
    midnight: "mi",
    noon: "n",
    morning: "morning",
    afternoon: "afternoon",
    evening: "evening",
    night: "night"
  },
  abbreviated: {
    am: "AM",
    pm: "PM",
    midnight: "midnight",
    noon: "noon",
    morning: "morning",
    afternoon: "afternoon",
    evening: "evening",
    night: "night"
  },
  wide: {
    am: "a.m.",
    pm: "p.m.",
    midnight: "midnight",
    noon: "noon",
    morning: "morning",
    afternoon: "afternoon",
    evening: "evening",
    night: "night"
  }
}, ir = {
  narrow: {
    am: "a",
    pm: "p",
    midnight: "mi",
    noon: "n",
    morning: "in the morning",
    afternoon: "in the afternoon",
    evening: "in the evening",
    night: "at night"
  },
  abbreviated: {
    am: "AM",
    pm: "PM",
    midnight: "midnight",
    noon: "noon",
    morning: "in the morning",
    afternoon: "in the afternoon",
    evening: "in the evening",
    night: "at night"
  },
  wide: {
    am: "a.m.",
    pm: "p.m.",
    midnight: "midnight",
    noon: "noon",
    morning: "in the morning",
    afternoon: "in the afternoon",
    evening: "in the evening",
    night: "at night"
  }
}, or = function(e, t) {
  var r = Number(e), n = r % 100;
  if (n > 20 || n < 10)
    switch (n % 10) {
      case 1:
        return r + "st";
      case 2:
        return r + "nd";
      case 3:
        return r + "rd";
    }
  return r + "th";
}, sr = {
  ordinalNumber: or,
  era: ne({
    values: er,
    defaultWidth: "wide"
  }),
  quarter: ne({
    values: tr,
    defaultWidth: "wide",
    argumentCallback: function(e) {
      return e - 1;
    }
  }),
  month: ne({
    values: rr,
    defaultWidth: "wide"
  }),
  day: ne({
    values: nr,
    defaultWidth: "wide"
  }),
  dayPeriod: ne({
    values: ar,
    defaultWidth: "wide",
    formattingValues: ir,
    defaultFormattingWidth: "wide"
  })
};
const ur = sr;
function ae(e) {
  return function(t) {
    var r = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, n = r.width, a = n && e.matchPatterns[n] || e.matchPatterns[e.defaultMatchWidth], i = t.match(a);
    if (!i)
      return null;
    var o = i[0], s = n && e.parsePatterns[n] || e.parsePatterns[e.defaultParseWidth], u = Array.isArray(s) ? cr(s, function(m) {
      return m.test(o);
    }) : lr(s, function(m) {
      return m.test(o);
    }), c;
    c = e.valueCallback ? e.valueCallback(u) : u, c = r.valueCallback ? r.valueCallback(c) : c;
    var d = t.slice(o.length);
    return {
      value: c,
      rest: d
    };
  };
}
function lr(e, t) {
  for (var r in e)
    if (e.hasOwnProperty(r) && t(e[r]))
      return r;
}
function cr(e, t) {
  for (var r = 0; r < e.length; r++)
    if (t(e[r]))
      return r;
}
function dr(e) {
  return function(t) {
    var r = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : {}, n = t.match(e.matchPattern);
    if (!n)
      return null;
    var a = n[0], i = t.match(e.parsePattern);
    if (!i)
      return null;
    var o = e.valueCallback ? e.valueCallback(i[0]) : i[0];
    o = r.valueCallback ? r.valueCallback(o) : o;
    var s = t.slice(a.length);
    return {
      value: o,
      rest: s
    };
  };
}
var fr = /^(\d+)(th|st|nd|rd)?/i, mr = /\d+/i, hr = {
  narrow: /^(b|a)/i,
  abbreviated: /^(b\.?\s?c\.?|b\.?\s?c\.?\s?e\.?|a\.?\s?d\.?|c\.?\s?e\.?)/i,
  wide: /^(before christ|before common era|anno domini|common era)/i
}, vr = {
  any: [/^b/i, /^(a|c)/i]
}, gr = {
  narrow: /^[1234]/i,
  abbreviated: /^q[1234]/i,
  wide: /^[1234](th|st|nd|rd)? quarter/i
}, yr = {
  any: [/1/i, /2/i, /3/i, /4/i]
}, pr = {
  narrow: /^[jfmasond]/i,
  abbreviated: /^(jan|feb|mar|apr|may|jun|jul|aug|sep|oct|nov|dec)/i,
  wide: /^(january|february|march|april|may|june|july|august|september|october|november|december)/i
}, wr = {
  narrow: [/^j/i, /^f/i, /^m/i, /^a/i, /^m/i, /^j/i, /^j/i, /^a/i, /^s/i, /^o/i, /^n/i, /^d/i],
  any: [/^ja/i, /^f/i, /^mar/i, /^ap/i, /^may/i, /^jun/i, /^jul/i, /^au/i, /^s/i, /^o/i, /^n/i, /^d/i]
}, br = {
  narrow: /^[smtwf]/i,
  short: /^(su|mo|tu|we|th|fr|sa)/i,
  abbreviated: /^(sun|mon|tue|wed|thu|fri|sat)/i,
  wide: /^(sunday|monday|tuesday|wednesday|thursday|friday|saturday)/i
}, Tr = {
  narrow: [/^s/i, /^m/i, /^t/i, /^w/i, /^t/i, /^f/i, /^s/i],
  any: [/^su/i, /^m/i, /^tu/i, /^w/i, /^th/i, /^f/i, /^sa/i]
}, Dr = {
  narrow: /^(a|p|mi|n|(in the|at) (morning|afternoon|evening|night))/i,
  any: /^([ap]\.?\s?m\.?|midnight|noon|(in the|at) (morning|afternoon|evening|night))/i
}, Sr = {
  any: {
    am: /^a/i,
    pm: /^p/i,
    midnight: /^mi/i,
    noon: /^no/i,
    morning: /morning/i,
    afternoon: /afternoon/i,
    evening: /evening/i,
    night: /night/i
  }
}, xr = {
  ordinalNumber: dr({
    matchPattern: fr,
    parsePattern: mr,
    valueCallback: function(e) {
      return parseInt(e, 10);
    }
  }),
  era: ae({
    matchPatterns: hr,
    defaultMatchWidth: "wide",
    parsePatterns: vr,
    defaultParseWidth: "any"
  }),
  quarter: ae({
    matchPatterns: gr,
    defaultMatchWidth: "wide",
    parsePatterns: yr,
    defaultParseWidth: "any",
    valueCallback: function(e) {
      return e + 1;
    }
  }),
  month: ae({
    matchPatterns: pr,
    defaultMatchWidth: "wide",
    parsePatterns: wr,
    defaultParseWidth: "any"
  }),
  day: ae({
    matchPatterns: br,
    defaultMatchWidth: "wide",
    parsePatterns: Tr,
    defaultParseWidth: "any"
  }),
  dayPeriod: ae({
    matchPatterns: Dr,
    defaultMatchWidth: "any",
    parsePatterns: Sr,
    defaultParseWidth: "any"
  })
};
const Cr = xr;
var Mr = {
  code: "en-US",
  formatDistance: At,
  formatLong: zt,
  formatRelative: Zt,
  localize: ur,
  match: Cr,
  options: {
    weekStartsOn: 0,
    firstWeekContainsDate: 1
  }
};
const Pr = Mr;
function Or(e, t) {
  g(2, arguments);
  var r = E(t);
  return Nt(e, -r);
}
var Wr = 864e5;
function kr(e) {
  g(1, arguments);
  var t = x(e), r = t.getTime();
  t.setUTCMonth(0, 1), t.setUTCHours(0, 0, 0, 0);
  var n = t.getTime(), a = r - n;
  return Math.floor(a / Wr) + 1;
}
function we(e) {
  g(1, arguments);
  var t = 1, r = x(e), n = r.getUTCDay(), a = (n < t ? 7 : 0) + n - t;
  return r.setUTCDate(r.getUTCDate() - a), r.setUTCHours(0, 0, 0, 0), r;
}
function et(e) {
  g(1, arguments);
  var t = x(e), r = t.getUTCFullYear(), n = new Date(0);
  n.setUTCFullYear(r + 1, 0, 4), n.setUTCHours(0, 0, 0, 0);
  var a = we(n), i = new Date(0);
  i.setUTCFullYear(r, 0, 4), i.setUTCHours(0, 0, 0, 0);
  var o = we(i);
  return t.getTime() >= a.getTime() ? r + 1 : t.getTime() >= o.getTime() ? r : r - 1;
}
function Er(e) {
  g(1, arguments);
  var t = et(e), r = new Date(0);
  r.setUTCFullYear(t, 0, 4), r.setUTCHours(0, 0, 0, 0);
  var n = we(r);
  return n;
}
var _r = 6048e5;
function $r(e) {
  g(1, arguments);
  var t = x(e), r = we(t).getTime() - Er(t).getTime();
  return Math.round(r / _r) + 1;
}
function be(e, t) {
  g(1, arguments);
  var r = t || {}, n = r.locale, a = n && n.options && n.options.weekStartsOn, i = a == null ? 0 : E(a), o = r.weekStartsOn == null ? i : E(r.weekStartsOn);
  if (!(o >= 0 && o <= 6))
    throw new RangeError("weekStartsOn must be between 0 and 6 inclusively");
  var s = x(e), u = s.getUTCDay(), c = (u < o ? 7 : 0) + u - o;
  return s.setUTCDate(s.getUTCDate() - c), s.setUTCHours(0, 0, 0, 0), s;
}
function tt(e, t) {
  g(1, arguments);
  var r = x(e), n = r.getUTCFullYear(), a = t || {}, i = a.locale, o = i && i.options && i.options.firstWeekContainsDate, s = o == null ? 1 : E(o), u = a.firstWeekContainsDate == null ? s : E(a.firstWeekContainsDate);
  if (!(u >= 1 && u <= 7))
    throw new RangeError("firstWeekContainsDate must be between 1 and 7 inclusively");
  var c = new Date(0);
  c.setUTCFullYear(n + 1, 0, u), c.setUTCHours(0, 0, 0, 0);
  var d = be(c, t), m = new Date(0);
  m.setUTCFullYear(n, 0, u), m.setUTCHours(0, 0, 0, 0);
  var f = be(m, t);
  return r.getTime() >= d.getTime() ? n + 1 : r.getTime() >= f.getTime() ? n : n - 1;
}
function Ur(e, t) {
  g(1, arguments);
  var r = t || {}, n = r.locale, a = n && n.options && n.options.firstWeekContainsDate, i = a == null ? 1 : E(a), o = r.firstWeekContainsDate == null ? i : E(r.firstWeekContainsDate), s = tt(e, t), u = new Date(0);
  u.setUTCFullYear(s, 0, o), u.setUTCHours(0, 0, 0, 0);
  var c = be(u, t);
  return c;
}
var Nr = 6048e5;
function qr(e, t) {
  g(1, arguments);
  var r = x(e), n = be(r, t).getTime() - Ur(r, t).getTime();
  return Math.round(n / Nr) + 1;
}
function h(e, t) {
  for (var r = e < 0 ? "-" : "", n = Math.abs(e).toString(); n.length < t; )
    n = "0" + n;
  return r + n;
}
var Hr = {
  y: function(e, t) {
    var r = e.getUTCFullYear(), n = r > 0 ? r : 1 - r;
    return h(t === "yy" ? n % 100 : n, t.length);
  },
  M: function(e, t) {
    var r = e.getUTCMonth();
    return t === "M" ? String(r + 1) : h(r + 1, 2);
  },
  d: function(e, t) {
    return h(e.getUTCDate(), t.length);
  },
  a: function(e, t) {
    var r = e.getUTCHours() / 12 >= 1 ? "pm" : "am";
    switch (t) {
      case "a":
      case "aa":
        return r.toUpperCase();
      case "aaa":
        return r;
      case "aaaaa":
        return r[0];
      case "aaaa":
      default:
        return r === "am" ? "a.m." : "p.m.";
    }
  },
  h: function(e, t) {
    return h(e.getUTCHours() % 12 || 12, t.length);
  },
  H: function(e, t) {
    return h(e.getUTCHours(), t.length);
  },
  m: function(e, t) {
    return h(e.getUTCMinutes(), t.length);
  },
  s: function(e, t) {
    return h(e.getUTCSeconds(), t.length);
  },
  S: function(e, t) {
    var r = t.length, n = e.getUTCMilliseconds(), a = Math.floor(n * Math.pow(10, r - 3));
    return h(a, t.length);
  }
};
const R = Hr;
var Z = {
  am: "am",
  pm: "pm",
  midnight: "midnight",
  noon: "noon",
  morning: "morning",
  afternoon: "afternoon",
  evening: "evening",
  night: "night"
}, Fr = {
  G: function(e, t, r) {
    var n = e.getUTCFullYear() > 0 ? 1 : 0;
    switch (t) {
      case "G":
      case "GG":
      case "GGG":
        return r.era(n, {
          width: "abbreviated"
        });
      case "GGGGG":
        return r.era(n, {
          width: "narrow"
        });
      case "GGGG":
      default:
        return r.era(n, {
          width: "wide"
        });
    }
  },
  y: function(e, t, r) {
    if (t === "yo") {
      var n = e.getUTCFullYear(), a = n > 0 ? n : 1 - n;
      return r.ordinalNumber(a, {
        unit: "year"
      });
    }
    return R.y(e, t);
  },
  Y: function(e, t, r, n) {
    var a = tt(e, n), i = a > 0 ? a : 1 - a;
    if (t === "YY") {
      var o = i % 100;
      return h(o, 2);
    }
    return t === "Yo" ? r.ordinalNumber(i, {
      unit: "year"
    }) : h(i, t.length);
  },
  R: function(e, t) {
    var r = et(e);
    return h(r, t.length);
  },
  u: function(e, t) {
    var r = e.getUTCFullYear();
    return h(r, t.length);
  },
  Q: function(e, t, r) {
    var n = Math.ceil((e.getUTCMonth() + 1) / 3);
    switch (t) {
      case "Q":
        return String(n);
      case "QQ":
        return h(n, 2);
      case "Qo":
        return r.ordinalNumber(n, {
          unit: "quarter"
        });
      case "QQQ":
        return r.quarter(n, {
          width: "abbreviated",
          context: "formatting"
        });
      case "QQQQQ":
        return r.quarter(n, {
          width: "narrow",
          context: "formatting"
        });
      case "QQQQ":
      default:
        return r.quarter(n, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  q: function(e, t, r) {
    var n = Math.ceil((e.getUTCMonth() + 1) / 3);
    switch (t) {
      case "q":
        return String(n);
      case "qq":
        return h(n, 2);
      case "qo":
        return r.ordinalNumber(n, {
          unit: "quarter"
        });
      case "qqq":
        return r.quarter(n, {
          width: "abbreviated",
          context: "standalone"
        });
      case "qqqqq":
        return r.quarter(n, {
          width: "narrow",
          context: "standalone"
        });
      case "qqqq":
      default:
        return r.quarter(n, {
          width: "wide",
          context: "standalone"
        });
    }
  },
  M: function(e, t, r) {
    var n = e.getUTCMonth();
    switch (t) {
      case "M":
      case "MM":
        return R.M(e, t);
      case "Mo":
        return r.ordinalNumber(n + 1, {
          unit: "month"
        });
      case "MMM":
        return r.month(n, {
          width: "abbreviated",
          context: "formatting"
        });
      case "MMMMM":
        return r.month(n, {
          width: "narrow",
          context: "formatting"
        });
      case "MMMM":
      default:
        return r.month(n, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  L: function(e, t, r) {
    var n = e.getUTCMonth();
    switch (t) {
      case "L":
        return String(n + 1);
      case "LL":
        return h(n + 1, 2);
      case "Lo":
        return r.ordinalNumber(n + 1, {
          unit: "month"
        });
      case "LLL":
        return r.month(n, {
          width: "abbreviated",
          context: "standalone"
        });
      case "LLLLL":
        return r.month(n, {
          width: "narrow",
          context: "standalone"
        });
      case "LLLL":
      default:
        return r.month(n, {
          width: "wide",
          context: "standalone"
        });
    }
  },
  w: function(e, t, r, n) {
    var a = qr(e, n);
    return t === "wo" ? r.ordinalNumber(a, {
      unit: "week"
    }) : h(a, t.length);
  },
  I: function(e, t, r) {
    var n = $r(e);
    return t === "Io" ? r.ordinalNumber(n, {
      unit: "week"
    }) : h(n, t.length);
  },
  d: function(e, t, r) {
    return t === "do" ? r.ordinalNumber(e.getUTCDate(), {
      unit: "date"
    }) : R.d(e, t);
  },
  D: function(e, t, r) {
    var n = kr(e);
    return t === "Do" ? r.ordinalNumber(n, {
      unit: "dayOfYear"
    }) : h(n, t.length);
  },
  E: function(e, t, r) {
    var n = e.getUTCDay();
    switch (t) {
      case "E":
      case "EE":
      case "EEE":
        return r.day(n, {
          width: "abbreviated",
          context: "formatting"
        });
      case "EEEEE":
        return r.day(n, {
          width: "narrow",
          context: "formatting"
        });
      case "EEEEEE":
        return r.day(n, {
          width: "short",
          context: "formatting"
        });
      case "EEEE":
      default:
        return r.day(n, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  e: function(e, t, r, n) {
    var a = e.getUTCDay(), i = (a - n.weekStartsOn + 8) % 7 || 7;
    switch (t) {
      case "e":
        return String(i);
      case "ee":
        return h(i, 2);
      case "eo":
        return r.ordinalNumber(i, {
          unit: "day"
        });
      case "eee":
        return r.day(a, {
          width: "abbreviated",
          context: "formatting"
        });
      case "eeeee":
        return r.day(a, {
          width: "narrow",
          context: "formatting"
        });
      case "eeeeee":
        return r.day(a, {
          width: "short",
          context: "formatting"
        });
      case "eeee":
      default:
        return r.day(a, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  c: function(e, t, r, n) {
    var a = e.getUTCDay(), i = (a - n.weekStartsOn + 8) % 7 || 7;
    switch (t) {
      case "c":
        return String(i);
      case "cc":
        return h(i, t.length);
      case "co":
        return r.ordinalNumber(i, {
          unit: "day"
        });
      case "ccc":
        return r.day(a, {
          width: "abbreviated",
          context: "standalone"
        });
      case "ccccc":
        return r.day(a, {
          width: "narrow",
          context: "standalone"
        });
      case "cccccc":
        return r.day(a, {
          width: "short",
          context: "standalone"
        });
      case "cccc":
      default:
        return r.day(a, {
          width: "wide",
          context: "standalone"
        });
    }
  },
  i: function(e, t, r) {
    var n = e.getUTCDay(), a = n === 0 ? 7 : n;
    switch (t) {
      case "i":
        return String(a);
      case "ii":
        return h(a, t.length);
      case "io":
        return r.ordinalNumber(a, {
          unit: "day"
        });
      case "iii":
        return r.day(n, {
          width: "abbreviated",
          context: "formatting"
        });
      case "iiiii":
        return r.day(n, {
          width: "narrow",
          context: "formatting"
        });
      case "iiiiii":
        return r.day(n, {
          width: "short",
          context: "formatting"
        });
      case "iiii":
      default:
        return r.day(n, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  a: function(e, t, r) {
    var n = e.getUTCHours(), a = n / 12 >= 1 ? "pm" : "am";
    switch (t) {
      case "a":
      case "aa":
        return r.dayPeriod(a, {
          width: "abbreviated",
          context: "formatting"
        });
      case "aaa":
        return r.dayPeriod(a, {
          width: "abbreviated",
          context: "formatting"
        }).toLowerCase();
      case "aaaaa":
        return r.dayPeriod(a, {
          width: "narrow",
          context: "formatting"
        });
      case "aaaa":
      default:
        return r.dayPeriod(a, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  b: function(e, t, r) {
    var n = e.getUTCHours(), a;
    switch (n === 12 ? a = Z.noon : n === 0 ? a = Z.midnight : a = n / 12 >= 1 ? "pm" : "am", t) {
      case "b":
      case "bb":
        return r.dayPeriod(a, {
          width: "abbreviated",
          context: "formatting"
        });
      case "bbb":
        return r.dayPeriod(a, {
          width: "abbreviated",
          context: "formatting"
        }).toLowerCase();
      case "bbbbb":
        return r.dayPeriod(a, {
          width: "narrow",
          context: "formatting"
        });
      case "bbbb":
      default:
        return r.dayPeriod(a, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  B: function(e, t, r) {
    var n = e.getUTCHours(), a;
    switch (n >= 17 ? a = Z.evening : n >= 12 ? a = Z.afternoon : n >= 4 ? a = Z.morning : a = Z.night, t) {
      case "B":
      case "BB":
      case "BBB":
        return r.dayPeriod(a, {
          width: "abbreviated",
          context: "formatting"
        });
      case "BBBBB":
        return r.dayPeriod(a, {
          width: "narrow",
          context: "formatting"
        });
      case "BBBB":
      default:
        return r.dayPeriod(a, {
          width: "wide",
          context: "formatting"
        });
    }
  },
  h: function(e, t, r) {
    if (t === "ho") {
      var n = e.getUTCHours() % 12;
      return n === 0 && (n = 12), r.ordinalNumber(n, {
        unit: "hour"
      });
    }
    return R.h(e, t);
  },
  H: function(e, t, r) {
    return t === "Ho" ? r.ordinalNumber(e.getUTCHours(), {
      unit: "hour"
    }) : R.H(e, t);
  },
  K: function(e, t, r) {
    var n = e.getUTCHours() % 12;
    return t === "Ko" ? r.ordinalNumber(n, {
      unit: "hour"
    }) : h(n, t.length);
  },
  k: function(e, t, r) {
    var n = e.getUTCHours();
    return n === 0 && (n = 24), t === "ko" ? r.ordinalNumber(n, {
      unit: "hour"
    }) : h(n, t.length);
  },
  m: function(e, t, r) {
    return t === "mo" ? r.ordinalNumber(e.getUTCMinutes(), {
      unit: "minute"
    }) : R.m(e, t);
  },
  s: function(e, t, r) {
    return t === "so" ? r.ordinalNumber(e.getUTCSeconds(), {
      unit: "second"
    }) : R.s(e, t);
  },
  S: function(e, t) {
    return R.S(e, t);
  },
  X: function(e, t, r, n) {
    var a = n._originalDate || e, i = a.getTimezoneOffset();
    if (i === 0)
      return "Z";
    switch (t) {
      case "X":
        return $e(i);
      case "XXXX":
      case "XX":
        return V(i);
      case "XXXXX":
      case "XXX":
      default:
        return V(i, ":");
    }
  },
  x: function(e, t, r, n) {
    var a = n._originalDate || e, i = a.getTimezoneOffset();
    switch (t) {
      case "x":
        return $e(i);
      case "xxxx":
      case "xx":
        return V(i);
      case "xxxxx":
      case "xxx":
      default:
        return V(i, ":");
    }
  },
  O: function(e, t, r, n) {
    var a = n._originalDate || e, i = a.getTimezoneOffset();
    switch (t) {
      case "O":
      case "OO":
      case "OOO":
        return "GMT" + _e(i, ":");
      case "OOOO":
      default:
        return "GMT" + V(i, ":");
    }
  },
  z: function(e, t, r, n) {
    var a = n._originalDate || e, i = a.getTimezoneOffset();
    switch (t) {
      case "z":
      case "zz":
      case "zzz":
        return "GMT" + _e(i, ":");
      case "zzzz":
      default:
        return "GMT" + V(i, ":");
    }
  },
  t: function(e, t, r, n) {
    var a = n._originalDate || e, i = Math.floor(a.getTime() / 1e3);
    return h(i, t.length);
  },
  T: function(e, t, r, n) {
    var a = n._originalDate || e, i = a.getTime();
    return h(i, t.length);
  }
};
function _e(e, t) {
  var r = e > 0 ? "-" : "+", n = Math.abs(e), a = Math.floor(n / 60), i = n % 60;
  if (i === 0)
    return r + String(a);
  var o = t || "";
  return r + String(a) + o + h(i, 2);
}
function $e(e, t) {
  if (e % 60 === 0) {
    var r = e > 0 ? "-" : "+";
    return r + h(Math.abs(e) / 60, 2);
  }
  return V(e, t);
}
function V(e, t) {
  var r = t || "", n = e > 0 ? "-" : "+", a = Math.abs(e), i = h(Math.floor(a / 60), 2), o = h(a % 60, 2);
  return n + i + r + o;
}
const Yr = Fr;
function Ue(e, t) {
  switch (e) {
    case "P":
      return t.date({
        width: "short"
      });
    case "PP":
      return t.date({
        width: "medium"
      });
    case "PPP":
      return t.date({
        width: "long"
      });
    case "PPPP":
    default:
      return t.date({
        width: "full"
      });
  }
}
function rt(e, t) {
  switch (e) {
    case "p":
      return t.time({
        width: "short"
      });
    case "pp":
      return t.time({
        width: "medium"
      });
    case "ppp":
      return t.time({
        width: "long"
      });
    case "pppp":
    default:
      return t.time({
        width: "full"
      });
  }
}
function Rr(e, t) {
  var r = e.match(/(P+)(p+)?/) || [], n = r[1], a = r[2];
  if (!a)
    return Ue(e, t);
  var i;
  switch (n) {
    case "P":
      i = t.dateTime({
        width: "short"
      });
      break;
    case "PP":
      i = t.dateTime({
        width: "medium"
      });
      break;
    case "PPP":
      i = t.dateTime({
        width: "long"
      });
      break;
    case "PPPP":
    default:
      i = t.dateTime({
        width: "full"
      });
      break;
  }
  return i.replace("{{date}}", Ue(n, t)).replace("{{time}}", rt(a, t));
}
var Ir = {
  p: rt,
  P: Rr
};
const Lr = Ir;
var Br = ["D", "DD"], jr = ["YY", "YYYY"];
function Ar(e) {
  return Br.indexOf(e) !== -1;
}
function Vr(e) {
  return jr.indexOf(e) !== -1;
}
function Ne(e, t, r) {
  if (e === "YYYY")
    throw new RangeError("Use `yyyy` instead of `YYYY` (in `".concat(t, "`) for formatting years to the input `").concat(r, "`; see: https://git.io/fxCyr"));
  if (e === "YY")
    throw new RangeError("Use `yy` instead of `YY` (in `".concat(t, "`) for formatting years to the input `").concat(r, "`; see: https://git.io/fxCyr"));
  if (e === "D")
    throw new RangeError("Use `d` instead of `D` (in `".concat(t, "`) for formatting days of the month to the input `").concat(r, "`; see: https://git.io/fxCyr"));
  if (e === "DD")
    throw new RangeError("Use `dd` instead of `DD` (in `".concat(t, "`) for formatting days of the month to the input `").concat(r, "`; see: https://git.io/fxCyr"));
}
var Xr = /[yYQqMLwIdDecihHKkms]o|(\w)\1*|''|'(''|[^'])+('|$)|./g, Gr = /P+p+|P+|p+|''|'(''|[^'])+('|$)|./g, Qr = /^'([^]*?)'?$/, zr = /''/g, Jr = /[a-zA-Z]/;
function Te(e, t, r) {
  g(2, arguments);
  var n = String(t), a = r || {}, i = a.locale || Pr, o = i.options && i.options.firstWeekContainsDate, s = o == null ? 1 : E(o), u = a.firstWeekContainsDate == null ? s : E(a.firstWeekContainsDate);
  if (!(u >= 1 && u <= 7))
    throw new RangeError("firstWeekContainsDate must be between 1 and 7 inclusively");
  var c = i.options && i.options.weekStartsOn, d = c == null ? 0 : E(c), m = a.weekStartsOn == null ? d : E(a.weekStartsOn);
  if (!(m >= 0 && m <= 6))
    throw new RangeError("weekStartsOn must be between 0 and 6 inclusively");
  if (!i.localize)
    throw new RangeError("locale must contain localize property");
  if (!i.formatLong)
    throw new RangeError("locale must contain formatLong property");
  var f = x(e);
  if (!It(f))
    throw new RangeError("Invalid time value");
  var O = qt(f), C = Or(f, O), $ = {
    firstWeekContainsDate: u,
    weekStartsOn: m,
    locale: i,
    _originalDate: f
  }, N = n.match(Gr).map(function(v) {
    var D = v[0];
    if (D === "p" || D === "P") {
      var W = Lr[D];
      return W(v, i.formatLong, $);
    }
    return v;
  }).join("").match(Xr).map(function(v) {
    if (v === "''")
      return "'";
    var D = v[0];
    if (D === "'")
      return Kr(v);
    var W = Yr[D];
    if (W)
      return !a.useAdditionalWeekYearTokens && Vr(v) && Ne(v, t, e), !a.useAdditionalDayOfYearTokens && Ar(v) && Ne(v, t, e), W(C, v, i.localize, $);
    if (D.match(Jr))
      throw new RangeError("Format string contains an unescaped latin alphabet character `" + D + "`");
    return v;
  }).join("");
  return N;
}
function Kr(e) {
  return e.match(Qr)[1].replace(zr, "'");
}
function me(e) {
  g(1, arguments);
  var t = x(e), r = t.getTime();
  return r;
}
function nt(e) {
  return g(1, arguments), Yt(e, Date.now());
}
function Zr(e, t) {
  if (arguments.length < 1)
    throw new TypeError("1 argument required, but only none provided present");
  var r = t && "nearestTo" in t ? E(t.nearestTo) : 1;
  if (r < 1 || r > 30)
    throw new RangeError("`options.nearestTo` must be between 1 and 30");
  var n = x(e), a = n.getSeconds(), i = n.getMinutes() + a / 60, o = Math.floor(i / r) * r, s = i % r, u = Math.round(s / r) * r;
  return new Date(n.getFullYear(), n.getMonth(), n.getDate(), n.getHours(), o + u);
}
function Ce() {
  return L(Date.now());
}
const en = (e, ...t) => {
  const r = { ...e };
  for (const n of t)
    delete r[n];
  return r;
}, qe = (e) => new Array(e).fill(""), tn = (e) => {
  const { width: t, height: r, top: n, left: a } = e.position;
  return {
    ...e,
    position: { width: t, height: r, top: n, left: a }
  };
}, ga = (e) => e / Wt, k = (e) => new Date(e), he = (e) => Math.abs(e), rn = ({
  dayWidth: e,
  startDate: t,
  endDate: r
}) => {
  const n = k(t), a = k(r);
  a < n && console.error(`Invalid endDate property. Value of endDate must be greater than startDate. Props: startDateTime: ${n}, endDateTime: ${a}`);
  const i = Ee(n, L(n)), o = Ee(a, n), s = Math.floor(e / o), u = s * o;
  return {
    hourWidth: he(s),
    dayWidth: he(u),
    numberOfHoursInDay: he(o),
    offsetStartHoursRange: he(i)
  };
}, z = (e) => new Date(e).getTime(), He = (e, t) => {
  const r = z(new Date()), n = z(e), a = z(t);
  return r >= n && a > r;
}, K = (e) => Te(new Date(e), re.DEFAULT).replace(/\s/g, "T"), ie = (e) => Zr(new Date(e)), nn = (e, t) => {
  const r = z(new Date(e));
  return z(new Date(t)) > r;
}, an = (e) => {
  const t = z(new Date(e)), r = z(new Date());
  return t > r;
}, ya = (e, t) => {
  let r = t;
  return t === "" && (r = K(L(Je(new Date(e), 1)))), { startDate: e, endDate: S(r) };
};
function on(e) {
  return K(L(Je(new Date(e), 1)));
}
const ve = (e, t) => e * t / Qe, J = (e, t, r, n, a) => {
  const i = me(k(t)) > me(k(n)), o = me(k(e)) < me(k(r));
  if (o && i) {
    const u = fe(ie(k(n)), k(r));
    return ve(u, a);
  }
  if (o) {
    const u = fe(ie(k(t)), k(r));
    return ve(u, a);
  }
  if (i) {
    const u = fe(k(n), ie(k(e)));
    return u < 0 ? 0 : ve(u, a);
  }
  const s = fe(ie(k(t)), ie(k(e)));
  return ve(s, a);
}, sn = (e, t) => ({
  top: t * e,
  height: t
}), un = (e, t, r, n, a, i) => {
  const o = {
    ...e,
    since: K(e.since),
    till: K(e.till)
  }, s = nn(o.since, a);
  let u = J(o.since, o.till, a, i, n);
  const c = r * t;
  let d = J(a, o.since, a, i, n);
  const m = J(a, o.till, a, i, n);
  return s && (d = 0), c < 0 && (u = 0), { position: {
    width: u,
    height: r,
    top: c,
    left: d,
    edgeEnd: m
  }, data: o };
}, ln = ({
  data: e,
  channels: t,
  startDate: r,
  endDate: n,
  itemHeight: a,
  hourWidth: i
}) => e.map((o) => {
  const s = t.findIndex(({ uuid: u }) => u === o.channelUuid);
  return un(o, s, a, i, r, n);
}, []), cn = (e, t) => e.map((r, n) => ({
  ...r,
  position: sn(n, t)
})), dn = (e, t, r, n, a, i) => e.width <= 0 || t > e.top + i * 3 || t + n <= e.top ? !1 : r + a >= e.left, fn = (e, t, r, n) => !(t > e.top + n * 3 || t + r <= e.top);
var Fe;
const Me = typeof window < "u";
Me && ((Fe = window == null ? void 0 : window.navigator) == null ? void 0 : Fe.userAgent) && /iP(ad|hone|od)/.test(window.navigator.userAgent);
function at(e, t) {
  function r(...n) {
    e(() => t.apply(this, n), { fn: t, thisArg: this, args: n });
  }
  return r;
}
const mn = (e) => e();
function hn(e, t = {}) {
  let r, n;
  return (i) => {
    const o = se(e), s = se(t.maxWait);
    if (r && clearTimeout(r), o <= 0 || s !== void 0 && s <= 0)
      return n && (clearTimeout(n), n = null), i();
    s && !n && (n = setTimeout(() => {
      r && clearTimeout(r), n = null, i();
    }, s)), r = setTimeout(() => {
      n && clearTimeout(n), n = null, i();
    }, o);
  };
}
function vn(e, t = !0, r = !0) {
  let n = 0, a, i = !0;
  const o = () => {
    a && (clearTimeout(a), a = void 0);
  };
  return (u) => {
    const c = se(e), d = Date.now() - n;
    if (o(), c <= 0)
      return n = Date.now(), u();
    d > c && (r || !i) ? (n = Date.now(), u()) : t && (a = setTimeout(() => {
      n = Date.now(), i = !0, o(), u();
    }, c)), !r && !a && (a = setTimeout(() => i = !0, c)), i = !1;
  };
}
function gn(e = "this function") {
  if (!Pt)
    throw new Error(`[VueUse] ${e} is only works on Vue 3.`);
}
function yn(e) {
  const t = Symbol("InjectionState");
  return [(...a) => {
    Ct(t, e(...a));
  }, () => Mt(t)];
}
function Ye(e) {
  return Dt() ? (St(e), !0) : !1;
}
function pn(e, t, { enumerable: r = !1, unwrap: n = !0 } = {}) {
  gn();
  for (const [a, i] of Object.entries(t))
    a !== "value" && (Xe(i) && n ? Object.defineProperty(e, a, {
      get() {
        return i.value;
      },
      set(o) {
        i.value = o;
      },
      enumerable: r
    }) : Object.defineProperty(e, a, { value: i, enumerable: r }));
  return e;
}
function wn(e, t = 200, r = {}) {
  return at(hn(t, r), e);
}
function bn(e, t = {}) {
  let r = e, n, a;
  const i = xt((f, O) => (n = f, a = O, {
    get() {
      return o();
    },
    set(C) {
      s(C);
    }
  }));
  function o(f = !0) {
    return f && n(), r;
  }
  function s(f, O = !0) {
    var C, $;
    if (f === r)
      return;
    const N = r;
    ((C = t.onBeforeChange) == null ? void 0 : C.call(t, f, N)) !== !1 && (r = f, ($ = t.onChanged) == null || $.call(t, f, N), O && a());
  }
  return pn(i, {
    get: o,
    set: s,
    untrackedGet: () => o(!1),
    silentSet: (f) => s(f, !1),
    peek: () => o(!1),
    lay: (f) => s(f, !1)
  }, { enumerable: !0 });
}
function X(e) {
  return typeof e == "function" ? Tt(e) : Ve(e);
}
function it(e, t = 1e3, r = {}) {
  const {
    immediate: n = !0,
    immediateCallback: a = !1
  } = r;
  let i = null;
  const o = Ve(!1);
  function s() {
    i && (clearInterval(i), i = null);
  }
  function u() {
    o.value = !1, s();
  }
  function c() {
    se(t) <= 0 || (o.value = !0, a && e(), s(), i = setInterval(e, se(t)));
  }
  if (n && Me && c(), Xe(t)) {
    const d = Ge(t, () => {
      o.value && Me && c();
    });
    Ye(d);
  }
  return Ye(u), {
    isActive: o,
    pause: u,
    resume: c
  };
}
var Re = Object.getOwnPropertySymbols, Tn = Object.prototype.hasOwnProperty, Dn = Object.prototype.propertyIsEnumerable, Sn = (e, t) => {
  var r = {};
  for (var n in e)
    Tn.call(e, n) && t.indexOf(n) < 0 && (r[n] = e[n]);
  if (e != null && Re)
    for (var n of Re(e))
      t.indexOf(n) < 0 && Dn.call(e, n) && (r[n] = e[n]);
  return r;
};
function xn(e, t, r = {}) {
  const n = r, {
    eventFilter: a = mn
  } = n, i = Sn(n, [
    "eventFilter"
  ]);
  return Ge(e, at(a, t), i);
}
var Cn = Object.defineProperty, Mn = Object.defineProperties, Pn = Object.getOwnPropertyDescriptors, De = Object.getOwnPropertySymbols, ot = Object.prototype.hasOwnProperty, st = Object.prototype.propertyIsEnumerable, Ie = (e, t, r) => t in e ? Cn(e, t, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[t] = r, On = (e, t) => {
  for (var r in t || (t = {}))
    ot.call(t, r) && Ie(e, r, t[r]);
  if (De)
    for (var r of De(t))
      st.call(t, r) && Ie(e, r, t[r]);
  return e;
}, Wn = (e, t) => Mn(e, Pn(t)), kn = (e, t) => {
  var r = {};
  for (var n in e)
    ot.call(e, n) && t.indexOf(n) < 0 && (r[n] = e[n]);
  if (e != null && De)
    for (var n of De(e))
      t.indexOf(n) < 0 && st.call(e, n) && (r[n] = e[n]);
  return r;
};
function En(e, t, r = {}) {
  const n = r, {
    throttle: a = 0,
    trailing: i = !0,
    leading: o = !0
  } = n, s = kn(n, [
    "throttle",
    "trailing",
    "leading"
  ]);
  return xn(e, t, Wn(On({}, s), {
    eventFilter: vn(a, i, o)
  }));
}
const [_n, $n] = yn((e) => {
  const t = X(e.theme), r = X(e.sidebarWidth), n = S();
  return { theme: t, sidebarWidth: r, scrollBoxRef: n };
});
function B() {
  const e = $n();
  if (e == null)
    throw new Error("Please call `useProvideEpgStore` on the appropriate parent component");
  return e;
}
const Un = /* @__PURE__ */ b("div", null, null, -1), Nn = [
  Un
], qn = /* @__PURE__ */ q({
  __name: "SideBar",
  setup(e) {
    const { theme: t, sidebarWidth: r } = B();
    return (n, a) => (p(), M("div", {
      class: "absolute z-900 top-0 left-0",
      style: T({
        width: `${l(r)}px`,
        height: `${l(kt)}px`,
        background: `${l(t).primary[900]}`
      })
    }, Nn, 4));
  }
}), H = (e, t) => {
  const r = e.__vccOpts || e;
  for (const [n, a] of t)
    r[n] = a;
  return r;
}, Hn = /* @__PURE__ */ H(qn, [["__file", "/Users/thudojsc/Dev/vue-epg-planby/src/Epg/components/SideBar.vue"]]), Fn = /* @__PURE__ */ b("svg", {
  class: "animate-spin mr-3 h-5 w-5",
  xmlns: "http://www.w3.org/2000/svg",
  fill: "none",
  viewBox: "0 0 24 24"
}, [
  /* @__PURE__ */ b("circle", {
    class: "opacity-25",
    cx: "12",
    cy: "12",
    r: "10",
    stroke: "currentColor",
    "stroke-width": "4"
  }),
  /* @__PURE__ */ b("path", {
    class: "opacity-75",
    fill: "currentColor",
    d: "M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
  })
], -1), Yn = /* @__PURE__ */ vt(" Processing... "), Rn = [
  Fn,
  Yn
], In = /* @__PURE__ */ q({
  __name: "Loader",
  setup(e) {
    const { theme: t } = B();
    return (r, n) => (p(), M("div", {
      class: "z-1000 absolute inset-0 flex items-center justify-center",
      style: T({
        background: `${l(t).loader.bg}`,
        color: `${l(t).white}`
      })
    }, Rn, 4));
  }
}), Ln = /* @__PURE__ */ H(In, [["__file", "/Users/thudojsc/Dev/vue-epg-planby/src/Epg/components/Loader.vue"]]), Bn = { class: "h-full w-full flex flex-col relative rounded overflow-hidden" }, jn = /* @__PURE__ */ q({
  __name: "Epg",
  props: {
    width: { type: Number, required: !1 },
    height: { type: Number, required: !1 },
    isSidebar: { type: Boolean, required: !0 },
    isTimeline: { type: Boolean, required: !1 },
    isLoading: { type: Boolean, required: !1 },
    theme: { type: null, required: !0 },
    sidebarWidth: { type: Number, required: !0 },
    setContainerRef: { type: Function, required: !0 }
  },
  setup(e) {
    return (t, r) => (p(), M("div", {
      ref: (n) => e.setContainerRef(n),
      style: T({ width: e.width ? `${e.width}px` : "100%", height: e.height ? `${e.height}px` : "100%" })
    }, [
      b("div", Bn, [
        e.isSidebar && e.isTimeline ? (p(), ee(Hn, { key: 0 })) : Q("v-if", !0),
        e.isLoading ? U(t.$slots, "loader", { key: 1 }, () => [
          gt(Ln)
        ]) : Q("v-if", !0),
        U(t.$slots, "default")
      ])
    ], 4));
  }
}), pa = /* @__PURE__ */ H(jn, [["__file", "/Users/thudojsc/Dev/vue-epg-planby/src/Epg/Epg.vue"]]), An = /* @__PURE__ */ q({
  __name: "Line",
  props: {
    height: { type: Number, required: !0 },
    startDate: { type: null, required: !0 },
    endDate: { type: null, required: !0 },
    dayWidth: { type: Number, required: !0 },
    hourWidth: { type: Number, required: !0 },
    sidebarWidth: { type: Number, required: !0 }
  },
  setup(e) {
    const t = e;
    function r({
      startDate: m,
      endDate: f,
      dayWidth: O,
      hourWidth: C,
      sidebarWidth: $
    }) {
      const N = J(L(new Date(m.value)), new Date(), m.value, f.value, C.value) + $.value, v = S(N), D = P(() => v.value <= O.value), W = P(() => D.value ? ze : void 0);
      return it(() => {
        const y = C.value / Qe * 2;
        v.value += y;
      }, W), Be(() => {
        const F = J(L(new Date(m.value)), new Date(), m.value, f.value, C.value) + $.value;
        v.value = F;
      }), { positionX: v };
    }
    const { startDate: n, endDate: a, dayWidth: i, hourWidth: o, sidebarWidth: s } = yt(t), { positionX: u } = r({
      startDate: n,
      endDate: a,
      dayWidth: i,
      hourWidth: o,
      sidebarWidth: s
    }), c = new Date(t.startDate), { theme: d } = B();
    return (m, f) => l(nt)(l(c)) ? (p(), M("div", {
      key: 0,
      class: "absolute top-64px w-3px pointer-events-none z-100",
      style: T({
        height: `${e.height}px`,
        left: `${l(u)}px`,
        background: `${l(d).green[300]}`
      })
    }, null, 4)) : Q("v-if", !0);
  }
}), Vn = /* @__PURE__ */ H(An, [["__file", "/Users/thudojsc/Dev/vue-epg-planby/src/Epg/components/Line.vue"]]);
function Xn({
  program: e,
  isBaseTimeFormat: t,
  minWidth: r = 200
}) {
  const { data: n, position: a } = e.value, { width: i } = a, { since: o, till: s } = n, u = S(He(o, s)), c = (f, O = re.HOURS_MIN) => Te(new Date(f), O).replace(/\s/g, ""), d = () => t.value ? re.BASE_HOURS_TIME : re.HOURS_MIN;
  it(() => {
    const f = He(o, s);
    u.value = f;
  }, ze);
  const m = i > r;
  return {
    format12HoursTime: (f) => c(f, d()),
    isLive: u,
    isMinWidth: m,
    styles: P(() => {
      const f = en(e.value.position, "egdeEnd");
      return {
        width: i,
        position: {
          width: `${f.width}px`,
          height: `${f.height}px`,
          top: `${f.top}px`,
          left: `${f.left}px`
        }
      };
    })
  };
}
function Gn(e, t) {
  const r = P(() => qe(e.value)), n = qe(4);
  return { times: r, dividers: n, formatTimelineTime: (i) => {
    const o = new Date(), s = Te(o, re.DATE), u = i < 10 ? `0${i}` : i;
    if (t.value) {
      const c = new Date(`${s}T${u}:00:00`);
      return Te(c, re.BASE_HOURS_TIME).toLowerCase().replace(/\s/g, "");
    }
    return `${u}:00`;
  } };
}
const Qn = { class: "h-full w-full grid grid-cols-4 items-end pb-6px" }, zn = { class: "text-14px relative" }, Jn = { class: "h-full w-full grid grid-cols-4 items-end pb-6px" }, Kn = /* @__PURE__ */ q({
  __name: "Timeline",
  props: {
    isBaseTimeFormat: { type: null, required: !0 },
    isSidebar: { type: Boolean, required: !0 },
    dayWidth: { type: Number, required: !0 },
    hourWidth: { type: Number, required: !0 },
    numberOfHoursInDay: { type: Number, required: !0 },
    offsetStartHoursRange: { type: Number, required: !0 },
    sidebarWidth: { type: Number, required: !0 }
  },
  setup(e) {
    const t = e, { theme: r } = B(), n = Gn(ge(t, "numberOfHoursInDay"), ge(t, "isBaseTimeFormat")), { times: a, dividers: i, formatTimelineTime: o } = n;
    return console.log("[LOG] ~ file: Timeline.vue ~ line 25 ~ times", a), (s, u) => (p(), M("div", {
      class: "sticky top-0 z-100 flex",
      style: T({
        left: `${e.isSidebar ? e.sidebarWidth : 0}px`,
        height: `${l(Pe) - 20}px`,
        width: `${e.dayWidth}px`,
        background: `${l(r).primary[900]}`
      })
    }, [
      (p(!0), M(ye, null, pe(l(a), (c, d) => U(s.$slots, "timeline", I(je({ key: d }, { index: d, dividers: l(i), formatTimelineTime: l(o), offsetStartHoursRange: e.offsetStartHoursRange, hourWidth: e.hourWidth, theme: l(r) })), () => [
        b("div", {
          class: "text-14px relative",
          style: T({
            width: `${e.hourWidth}px`
          })
        }, [
          b("div", {
            class: "absolute top-18px",
            style: T({
              color: `${l(r).text.grey[300]}`,
              left: `${d === 0 ? 0 : -18}px`
            })
          }, te(l(o)(d + e.offsetStartHoursRange)), 5),
          b("div", Qn, [
            (p(!0), M(ye, null, pe(l(i), (m, f) => (p(), M("div", {
              key: f,
              style: T({
                background: `${l(r).timeline.divider.bg}`,
                height: "10px",
                width: "1px",
                marginRight: `${e.hourWidth}px`
              })
            }, null, 4))), 128))
          ])
        ], 4)
      ])), 128)),
      U(s.$slots, "timeline-last", I(G({ theme: l(r), formatTimelineTime: l(o), times: l(a), offsetStartHoursRange: e.offsetStartHoursRange })), () => [
        b("div", zn, [
          b("div", {
            class: "absolute top-18px -translate-x-100%",
            style: T({
              color: `${l(r).text.grey[300]}`,
              left: `${0}px`
            })
          }, te(l(o)(l(a).length + e.offsetStartHoursRange)), 5),
          b("div", Jn, [
            b("div", {
              style: T({
                background: `${l(r).timeline.divider.bg}`,
                height: "10px",
                width: "1px"
              })
            }, null, 4)
          ])
        ])
      ])
    ], 4));
  }
}), Zn = /* @__PURE__ */ H(Kn, [["__file", "/Users/thudojsc/Dev/vue-epg-planby/src/Epg/components/Timeline.vue"]]), ea = ["src"], ta = /* @__PURE__ */ q({
  __name: "Channel",
  props: {
    channel: { type: null, required: !0 }
  },
  emits: ["click"],
  setup(e, { emit: t }) {
    const { theme: r } = B();
    return (n, a) => (p(), M("div", {
      "data-testid": "sidebar-item",
      class: "absolute flex items-center justify-center",
      style: T({
        top: `${e.channel.position.top}px`,
        height: `${e.channel.position.height}px`,
        width: "100%",
        backgroundColor: `${l(r).primary[900]}`
      }),
      onClick: a[0] || (a[0] = (i) => t("click", e.channel))
    }, [
      U(n.$slots, "channel", I(G({ channel: e.channel, theme: l(r) })), () => [
        b("img", {
          src: e.channel.logo,
          class: "max-h-60px max-w-60px relative"
        }, null, 8, ea)
      ])
    ], 4));
  }
}), ra = /* @__PURE__ */ H(ta, [["__file", "/Users/thudojsc/Dev/vue-epg-planby/src/Epg/components/Channel.vue"]]), na = /* @__PURE__ */ q({
  __name: "Channels",
  props: {
    channels: { type: Array, required: !0 },
    scrollY: { type: Number, required: !0 },
    sidebarWidth: { type: Number, required: !0 },
    isTimeline: { type: Boolean, required: !0 },
    isChannelVisible: { type: Function, required: !0 }
  },
  emits: ["click"],
  setup(e, { emit: t }) {
    const r = e, { theme: n } = B(), a = P(() => r.channels.filter((i) => r.isChannelVisible(i)));
    return (i, o) => (p(), M("div", {
      "data-testid": "sidebar",
      class: "sticky float-left",
      style: T({
        width: `${e.sidebarWidth}px`,
        bottom: `${e.scrollY}px`,
        left: 0,
        zIndex: 100,
        backgroundColor: `${l(n).primary[900]}`
      })
    }, [
      (p(!0), M(ye, null, pe(l(a), (s) => (p(), ee(ra, {
        key: s.uuid,
        channel: s,
        onClick: o[0] || (o[0] = (u) => t("click", u))
      }, {
        channel: oe((u) => [
          U(i.$slots, "channel", I(G(u)))
        ]),
        _: 2
      }, 1032, ["channel"]))), 128))
    ], 4));
  }
}), aa = /* @__PURE__ */ H(na, [["__file", "/Users/thudojsc/Dev/vue-epg-planby/src/Epg/components/Channels.vue"]]), ia = { class: "flex w-full justify-start" }, oa = ["src"], sa = { class: "overflow-hidden" }, ua = /* @__PURE__ */ q({
  __name: "Program",
  props: {
    program: { type: null, required: !0 },
    isBaseTimeFormat: { type: null, required: !0 }
  },
  emits: ["click"],
  setup(e, { emit: t }) {
    const r = e;
    Ae((d) => ({
      "dc18c6b7-theme_primary_600_": l(c).primary[600],
      "dc18c6b7-theme_gradient_blue_900_": l(c).gradient.blue[900],
      "dc18c6b7-theme_gradient_blue_600_": l(c).gradient.blue[600],
      "dc18c6b7-theme_gradient_blue_300_": l(c).gradient.blue[300]
    }));
    const n = P(() => r.program.data), a = Xn({
      program: ge(r, "program"),
      isBaseTimeFormat: ge(r, "isBaseTimeFormat")
    }), { styles: i, format12HoursTime: o, isLive: s, isMinWidth: u } = a, { theme: c } = B();
    return (d, m) => (p(), M("div", {
      "data-testid": "program-item",
      class: "absolute overflow-hidden",
      style: T({
        ...l(i).position,
        padding: `${l(i).width === 0 ? 0 : 4}px`
      }),
      onClick: m[0] || (m[0] = (f) => t("click", l(n)))
    }, [
      U(d.$slots, "program", je({ ...l(a), theme: l(c) }, { data: l(n) }), () => [
        b("div", {
          "data-testid": "program-content",
          class: pt(["program-item cursor-pointer relative flex text-11px h-full rounded-8px overflow-hidden transition-all duration-400 ease-in-out z-1", {
            "is-live": l(s)
          }]),
          style: T({
            padding: `10px ${l(i).width < 30 ? 4 : 20}px`
          })
        }, [
          b("div", ia, [
            l(s) && l(u) ? (p(), M("img", {
              key: 0,
              class: "mr-15px rounded-6px w-100px",
              src: l(n).image,
              alt: "Preview"
            }, null, 8, oa)) : Q("v-if", !0),
            b("div", sa, [
              b("div", {
                class: "text-14px font-medium text-left mt-0 mb-5px truncate",
                style: T({
                  color: `${l(c).grey[300]}`
                })
              }, te(l(n).title), 5),
              b("div", {
                class: "block text-12.5px font-normal text-left truncate",
                style: T({
                  color: `${l(c).text.grey[500]}`
                }),
                "aria-label": "program time"
              }, te(l(o)(l(n).since)) + " -" + te(" ") + " " + te(l(o)(l(n).till)), 5)
            ])
          ])
        ], 6)
      ])
    ], 4));
  }
});
const la = /* @__PURE__ */ H(ua, [["__file", "/Users/thudojsc/Dev/vue-epg-planby/src/Epg/components/Program.vue"]]), ca = /* @__PURE__ */ q({
  __name: "Layout",
  props: {
    programs: { type: Array, required: !0 },
    channels: { type: Array, required: !0 },
    startDate: { type: null, required: !0 },
    endDate: { type: null, required: !0 },
    scrollY: { type: Number, required: !0 },
    dayWidth: { type: Number, required: !0 },
    hourWidth: { type: Number, required: !0 },
    numberOfHoursInDay: { type: Number, required: !0 },
    offsetStartHoursRange: { type: Number, required: !0 },
    sidebarWidth: { type: Number, required: !0 },
    itemHeight: { type: Number, required: !0 },
    onScroll: { type: Function, required: !0 },
    isBaseTimeFormat: { type: null, required: !1, default: !1 },
    isSidebar: { type: Boolean, required: !1, default: !0 },
    isTimeline: { type: Boolean, required: !1, default: !0 },
    isLine: { type: Boolean, required: !1, default: !0 },
    isProgramVisible: { type: Function, required: !0 },
    isChannelVisible: { type: Function, required: !0 },
    setScrollBoxRef: { type: Function, required: !0 }
  },
  emits: ["channel-click", "program-click"],
  setup(e, { emit: t }) {
    const r = e;
    Ae((u) => ({
      "92676bd7-theme_scrollbar_thumb_bg": l(o).scrollbar.thumb.bg,
      "92676bd7-theme_white": l(o).white,
      "92676bd7-theme_primary__900__": l(o).primary[900]
    }));
    const n = P(() => r.channels.length), a = P(() => n.value * r.itemHeight), i = P(() => an(r.endDate)), { theme: o } = B(), s = P(() => r.programs.filter((u) => r.isProgramVisible(u.position)));
    return (u, c) => (p(), M("div", {
      ref: (d) => e.setScrollBoxRef(d),
      class: "w-full h-full relative overflow-auto scroll-smooth scrollbox",
      style: T({
        background: `${l(o).primary[900]}`
      }),
      onScroll: c[1] || (c[1] = (...d) => e.onScroll && e.onScroll(...d))
    }, [
      e.isLine && l(i) ? (p(), ee(Vn, {
        key: 0,
        "day-width": e.dayWidth,
        "hour-width": e.hourWidth,
        "start-date": e.startDate,
        "sidebar-width": e.sidebarWidth,
        "end-date": e.endDate,
        height: l(a)
      }, null, 8, ["day-width", "hour-width", "start-date", "sidebar-width", "end-date", "height"])) : Q("v-if", !0),
      e.isTimeline ? (p(), ee(Zn, {
        key: 1,
        "is-base-time-format": e.isBaseTimeFormat,
        "is-sidebar": e.isSidebar,
        "day-width": e.dayWidth,
        "hour-width": e.hourWidth,
        "number-of-hours-in-day": e.numberOfHoursInDay,
        "offset-start-hours-range": e.offsetStartHoursRange,
        "sidebar-width": e.sidebarWidth
      }, {
        timeline: oe((d) => [
          U(u.$slots, "timeline", I(G(d)))
        ]),
        "timeline-last": oe((d) => [
          U(u.$slots, "timeline-last", I(G(d)))
        ]),
        _: 3
      }, 8, ["is-base-time-format", "is-sidebar", "day-width", "hour-width", "number-of-hours-in-day", "offset-start-hours-range", "sidebar-width"])) : Q("v-if", !0),
      e.isSidebar ? (p(), ee(aa, {
        key: 2,
        "sidebar-width": e.sidebarWidth,
        "is-timeline": e.isTimeline,
        "is-channel-visible": e.isChannelVisible,
        channels: e.channels,
        "scroll-y": e.scrollY,
        onClick: c[0] || (c[0] = (d) => t("channel-click", d))
      }, {
        channel: oe((d) => [
          U(u.$slots, "channel", I(G(d)))
        ]),
        _: 3
      }, 8, ["sidebar-width", "is-timeline", "is-channel-visible", "channels", "scroll-y"])) : Q("v-if", !0),
      b("div", {
        "data-testid": "content",
        class: "relative",
        style: T({
          background: `${l(o).primary[900]}`,
          height: `${l(a)}px`,
          width: `${e.dayWidth}px`,
          left: `${e.isSidebar ? e.sidebarWidth : 0}px`
        })
      }, [
        (p(!0), M(ye, null, pe(l(s), (d) => (p(), ee(la, {
          key: d.data.id,
          program: l(tn)(d),
          "is-base-time-format": e.isBaseTimeFormat,
          "is-program-visible": e.isProgramVisible,
          onClick: (m) => t("program-click", m)
        }, {
          program: oe((m) => [
            U(u.$slots, "program", I(G(m)))
          ]),
          _: 2
        }, 1032, ["program", "is-base-time-format", "is-program-visible", "onClick"]))), 128))
      ], 4)
    ], 36));
  }
});
const wa = /* @__PURE__ */ H(ca, [["__file", "/Users/thudojsc/Dev/vue-epg-planby/src/Epg/components/Layout.vue"]]), da = {
  primary: {
    600: "#1a202c",
    900: "#171923"
  },
  grey: { 300: "#d1d1d1" },
  white: "#fff",
  green: {
    300: "#2C7A7B"
  },
  loader: {
    teal: "#5DDADB",
    purple: "#3437A2",
    pink: "#F78EB6",
    bg: "#171923db"
  },
  scrollbar: {
    border: "#ffffff",
    thumb: {
      bg: "#e1e1e1"
    }
  },
  gradient: {
    blue: {
      300: "#002eb3",
      600: "#002360",
      900: "#051937"
    }
  },
  text: {
    grey: {
      300: "#a0aec0",
      500: "#718096"
    }
  },
  timeline: {
    divider: {
      bg: "#718096"
    }
  }
};
function fa({
  height: e,
  width: t,
  startDate: r,
  endDate: n,
  hourWidth: a,
  sidebarWidth: i
}) {
  const o = S(null), s = S(null), u = S(0), c = S(0), d = bn(0), m = S(t), f = S(e), O = P(() => nt(new Date(r.value))), C = (y) => {
    var _;
    const w = { y: y.target.scrollTop, x: y.target.scrollLeft };
    u.value = w.y, c.value = w.x;
    const Y = ((t != null ? t : (_ = o.value) == null ? void 0 : _.clientWidth) - i.value) / 2;
    d.silentSet((w.x + Y) / a.value);
  }, $ = () => {
    var y;
    if ((s == null ? void 0 : s.value) && O.value) {
      const w = t != null ? t : (y = o.value) == null ? void 0 : y.clientWidth, j = new Date(), _ = J(Ce(), j, r.value, n.value, a.value) - w / 2 + i.value;
      s.value.scrollLeft = _;
    }
  }, N = () => {
    var ce, de;
    const j = ((t != null ? t : (ce = o.value) == null ? void 0 : ce.clientWidth) - i.value) / 2 / a.value, Y = d.untrackedGet() - j, _ = new Date(r.value), ue = Math.floor(Y), le = Y - ue;
    _.setHours(ue), _.setMinutes(Math.round(le * 60));
    const Se = J(Ce(), _, r.value, n.value, a.value);
    (de = s.value) == null || de.scrollTo({ left: Se, behavior: "auto" });
  };
  En([a], () => {
    N();
  }, { throttle: 500, trailing: !0, leading: !0 });
  const v = (y = a.value) => {
    if (s != null && s.value) {
      const w = s.value.scrollTop + y;
      s.value.scrollTop = w;
    }
  }, D = (y = a.value) => {
    if (s != null && s.value) {
      const w = s.value.scrollLeft + y;
      s.value.scrollLeft = w;
    }
  }, W = (y = a.value) => {
    if (s != null && s.value) {
      const w = s.value.scrollLeft - y;
      s.value.scrollLeft = w;
    }
  }, F = wn(() => {
    if ((o == null ? void 0 : o.value) && !t) {
      const y = o.value, { clientWidth: w } = y;
      m.value = w;
    }
  }, $t * 4, { maxWait: Ut * 4 });
  return Be(() => {
    if (o != null && o.value) {
      const y = o.value;
      if (!t) {
        const { clientWidth: w } = y;
        m.value = w;
      }
      if (!e) {
        const { clientHeight: w } = y;
        f.value = w;
      }
    }
  }), wt(() => {
    window.addEventListener("resize", F);
  }), bt(() => {
    window.removeEventListener("resize", F);
  }), {
    containerRef: o,
    scrollBoxRef: s,
    scrollX: c,
    scrollY: u,
    layoutWidth: m,
    layoutHeight: f,
    onScroll: C,
    onScrollToNow: $,
    onScrollTop: v,
    onScrollLeft: W,
    onScrollRight: D
  };
}
const Le = K(Ce());
function ba({
  channels: e,
  epg: t,
  startDate: r = Le,
  endDate: n = on(Le),
  isBaseTimeFormat: a = S(!1),
  isSidebar: i = S(!0),
  isTimeline: o = S(!0),
  isLine: s = !0,
  theme: u,
  dayWidth: c = S(Ot),
  sidebarWidth: d = S(Et),
  itemHeight: m = S(Pe),
  itemOverscan: f = _t,
  width: O,
  height: C
}) {
  const $ = X(e), N = X(t), v = X(r), D = X(n), W = P(() => rn({
    dayWidth: c.value,
    startDate: v.value,
    endDate: D.value
  })), F = P(() => W.value.hourWidth), { containerRef: y, scrollBoxRef: w, ...j } = fa({
    startDate: v,
    endDate: D,
    sidebarWidth: d,
    width: O,
    height: C,
    hourWidth: F
  }), { scrollX: Y, scrollY: _, layoutWidth: ue, layoutHeight: le } = j, { onScroll: Se, onScrollToNow: ce, onScrollTop: de, onScrollLeft: ut, onScrollRight: lt } = j, Oe = P(() => cn($.value, m.value)), ct = P(() => ln({
    data: N.value,
    channels: Oe.value,
    startDate: K(v.value),
    endDate: K(D.value),
    itemHeight: m.value,
    hourWidth: F.value
  })), We = X(u || da), dt = (A) => dn(A, _.value, Y.value, le.value, ue.value, f), ft = (A) => fn(A, _.value, le.value, f), mt = () => ({
    width: O,
    height: C,
    isSidebar: i.value,
    isTimeline: o.value,
    sidebarWidth: d.value,
    theme: l(We),
    setContainerRef: (A) => y.value = A
  }), ht = () => ({
    programs: ct.value,
    channels: Oe.value,
    startDate: v.value,
    endDate: D.value,
    scrollY: _.value,
    onScroll: Se,
    isBaseTimeFormat: a.value,
    isSidebar: i.value,
    isTimeline: o.value,
    isLine: s,
    isProgramVisible: dt,
    isChannelVisible: ft,
    sidebarWidth: d.value,
    itemHeight: m.value,
    hourWidth: W.value.hourWidth,
    dayWidth: W.value.dayWidth,
    numberOfHoursInDay: W.value.numberOfHoursInDay,
    offsetStartHoursRange: W.value.offsetStartHoursRange,
    setScrollBoxRef: (A) => w.value = A
  });
  return _n({
    theme: We,
    sidebarWidth: d
  }), {
    getEpgProps: mt,
    getLayoutProps: ht,
    onScrollToNow: ce,
    onScrollTop: de,
    onScrollLeft: ut,
    onScrollRight: lt,
    scrollY: _,
    scrollX: Y
  };
}
export {
  Ot as DAY_WIDTH,
  $t as DEBOUNCE_WAIT,
  Ut as DEBOUNCE_WAIT_MAX,
  pa as Epg,
  Wt as HOURS_IN_DAY,
  Qe as HOUR_IN_MINUTES,
  Pe as ITEM_HEIGHT,
  _t as ITEM_OVERSCAN,
  wa as Layout,
  ze as PROGRAM_REFRESH,
  Et as SIDEBAR_WIDTH,
  va as THEME_MODE,
  kt as TIMELINE_HEIGHT,
  re as TIME_FORMAT,
  K as formatTime,
  qe as generateArray,
  sn as getChannelPosition,
  cn as getConvertedChannels,
  ln as getConvertedPrograms,
  k as getDate,
  rn as getDayWidthResources,
  on as getDefaultEndDate,
  ga as getHourWidth,
  dn as getItemVisibility,
  He as getLiveStatus,
  J as getPositionX,
  tn as getProgramOptions,
  un as getProgramPosition,
  fn as getSidebarItemVisibility,
  ya as getTimeRangeDates,
  an as isFutureTime,
  nn as isYesterday,
  en as omit,
  ie as roundToMinutes,
  ba as useEpg
};
